### ISSUE TITLE
USE_GEMMLOWP=ON option build and test errors

### ISSUE THREAD

**Issue by jnorwood** [tiny-dnn/tiny-dnn#977]
There were a couple of errors in the build.
params_f_  needed to be defined as a private variable in backend_tiny.h and initialized by the constructors.

min_a, min_b, max_a, max_b are undefined in tiny_quantized_matmul_kernel.h, but it doesn't appear to me that function is needed, since there is no float to quantized integer conversion in that file.   The file doesn't include the code to search for min and max value range from the inputs, which are integer.

after getting beyond those, the build goes ok, but testing fails.
Had to add a source for tanh() in test_quantized_fully_connected_layer, which I tried with:
using tanh = tiny_dnn::activation::tanh;

after that, looks like the current call to nn.gradient_check, from test_quantized_fully_connected_layer.h,  has an extra 1 parameter.  I haven't fixed that yet.

Was this all supposed to be working?

**Comment by edgarriba** [comment]
@jnorwood unfortunately quantization branch was never finished

**Comment by beru** [comment]
@edgarriba I suggest to get rid of the broken or unfinished code such as this and xtensor integration.

**Comment by edgarriba** [comment]
@beru agree. Let's do a clean up, sending not working code to graveyard could be a good approach.

**Comment by jnorwood** [comment]
Looks like someone did a lot of work to support that feature.  Was it never working?  It would be useful.

**Comment by beru** [comment]
@jnorwood 

The feature was implemeted with the support of GSoC 2016.
https://summerofcode.withgoogle.com/archive/2016/projects/4623962327744512/

But tiny-dnn's current tensor structure does not natively support integer values and `float` or `double` type is used to store values.

https://github.com/tiny-dnn/tiny-dnn/blob/a8de26ad40d955908c5ec4fc946c3c67dd381c6c/tiny_dnn/util/util.h#L66

https://github.com/tiny-dnn/tiny-dnn/blob/a8de26ad40d955908c5ec4fc946c3c67dd381c6c/tiny_dnn/config.h#L88

If you look at the implementation, you'll find type conversions take place before and after of quantized computations which spoils efficiency of it.

https://github.com/tiny-dnn/tiny-dnn/blob/a8de26ad40d955908c5ec4fc946c3c67dd381c6c/tiny_dnn/core/kernels/tiny_quantized_conv2d_kernel.h#L38

https://github.com/tiny-dnn/tiny-dnn/blob/a8de26ad40d955908c5ec4fc946c3c67dd381c6c/tiny_dnn/core/kernels/tiny_quantized_conv2d_kernel.h#L146
